src/server/pps/server/api_server.go

```diff
@@ -4130,7 +4134,21 @@ func (a *apiServer) SetProjectDefaults(ctx context.Context, req *pps.SetProjectD
 	}
 
 	if err := a.txnEnv.WithWriteContext(ctx, func(ctx context.Context, txnCtx *txncontext.TransactionContext) error {
-		if err := a.projectDefaults.ReadWrite(txnCtx.SqlTx).Put(ctx, req.GetProject().String(), &ppsdb.ProjectDefaultsWrapper{Json: req.GetProjectDefaultsJson()}); err != nil {
+		// get old defaults in order to get created_by and created_at
+		projectDefaultsTxn := a.projectDefaults.ReadWrite(txnCtx.SqlTx)
+		// If auth is active, make caller the owner of these new project defaults.
+		var username string
+		if whoAmI, err := txnCtx.WhoAmI(); err == nil {
+			username = whoAmI.GetUsername()
+		} else if !errors.Is(err, auth.ErrNotActivated) {
+			return errors.Wrap(err, "could not get caller's username")
+		}
+		var new = &ppsdb.ProjectDefaultsWrapper{
```

Comment:
only weird bit of behavior is that the system may include project defaults that were created before this code and won't have created-by and created-at populated

Reply:
That’s correct — they are not available, so are blank. We follow the same pattern for e.g. PFS projects.